api/src/services/survey-block-service.ts

```diff
@@ -0,0 +1,89 @@
+import { IDBConnection } from '../database/db';
+import { SurveyBlock, SurveyBlockRecord, SurveyBlockRepository } from '../repositories/survey-block-repository';
+import { DBService } from './db-service';
+
+export class SurveyBlockService extends DBService {
+  surveyBlockRepository: SurveyBlockRepository;
+
+  constructor(connection: IDBConnection) {
+    super(connection);
+    this.surveyBlockRepository = new SurveyBlockRepository(connection);
+  }
+
+  /**
+   * Gets Block Survey Records for a given survey id
+   *
+   * @param {number} surveyId
+   * @return {*} {Promise<SurveyBlockRecord[]>}
+   * @returns
+   */
+  async getSurveyBlocksForSurveyId(surveyId: number): Promise<SurveyBlockRecord[]> {
+    return await this.surveyBlockRepository.getSurveyBlocksForSurveyId(surveyId);
+  }
+
+  /**
+   *  Deletes a survey block record.
+   *
+   * @param {number} surveyBlockId
+   * @return {*}  {Promise<SurveyBlockRecord>}
+   * @memberof SurveyBlockService
+   */
+  async deleteSurveyBlock(surveyBlockId: number): Promise<SurveyBlockRecord> {
+    return this.surveyBlockRepository.deleteSurveyBlockRecord(surveyBlockId);
+  }
+
+  /**
+   * Insert or Updates Survey Block Records based on the existence of a survey_block_id
+   *
+   * @param {number} surveyId
+   * @param {SurveyBlock[]} blocks
+   * @return {*} {Promise<void>}
+   * @memberof SurveyBlockService
+   */
+  async updateInsertSurveyBlocks(surveyId: number, blocks: SurveyBlock[]): Promise<void> {
+    const insertUpdate: Promise<any>[] = [];
+
+    blocks.forEach((item: SurveyBlock) => {
+      item.survey_id = surveyId;
```

Comment:
Is this assignment needed?

Reply:
yup. If the block being added is being added to a branch new survey we won't have the ID until the api request is being made.

Reply:
Ahh understood